zaza/openstack/charm_tests/magnum/tests.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+#
+# Copyright 2019 Canonical Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Encapsulate magnum testing."""
+import logging
+import json
+import os
+import subprocess
+from urllib import parse as urlparse
+
+import zaza.model
+import zaza.openstack.charm_tests.nova.utils as nova_utils
+import zaza.openstack.charm_tests.test_utils as test_utils
+import zaza.openstack.utilities.openstack as openstack_utils
+import zaza.charm_lifecycle.utils as charm_lifecycle_utils
+
+# Resource and name constants
+IMAGE_NAME = 'fedora-coreos'
+CLUSTER_NAME = 'hello_world'
+TEMPLATE_NAME = 'hello-world-template'
+RESOURCE_TYPE = 'server'
+TEMPLATES_PATH = 'files'
+FLAVOR_NAME = 'm1.small'
+
+
+class MagnumBasicDeployment(test_utils.OpenStackBaseTest):
+    """Encapsulate Magnum tests."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running magnum tests."""
+        super(MagnumBasicDeployment, cls).setUpClass()
+        cls.application = 'magnum'
+        cls.keystone_session = openstack_utils.get_overcloud_keystone_session()
+        cls.magnum_client = openstack_utils.get_magnum_session_client(
+            cls.keystone_session)
+        cls.heat_client = openstack_utils.get_heat_session_client(
+            cls.keystone_session)
+        cls.glance_client = openstack_utils.get_glance_session_client(
+            cls.keystone_session)
+
+
+    @property
+    def services(self):
+        """Return a list services for the selected OpenStack release.
+
+        :returns: List of services
+        :rtype: [str]
+        """
+        services = ['magnum-api', 'magnum-conductor']
+        return services
+
+
+    def test_410_magnum_cluster_create_delete(self):
+        """Create cluster, confirm nova compute resource, delete cluster."""
+        # Verify new image name
+        images_list = list(self.glance_client.images.list())
+        self.assertEqual(images_list[0].name, IMAGE_NAME,
+                         "glance image create failed or unexpected")
```

Comment:
The assertion error message should `Magnum image not found`, and not `create failed or unexpected`. We don't create any image here.